lib/simplifyExpression/basicsSearch/basicRules.js

```diff
@@ -0,0 +1,224 @@
+const {build, query} = require('math-nodes');
+const {rewriteNode} = require('math-rules');
+
+const ChangeTypes = require('../../ChangeTypes');
+const checks = require('../../checks');
+const defineRuleString = require('../../util/defineRuleString');
+const Node = require('../../node');
+
+// e.g. -(-3) -> 3
+const NEGATION = defineRuleString('--#a', '#a');
```

Comment:
Good call. I would go with your second suggestion of collapsing multiple steps where we deal with pedagogy.